### PR TITLE
Module definition update #144

### PR DESCRIPTION
**Pull request by agudek**
AxisScale and AxisFitFunction fields now must return d3.svg.axis instances. This gives module creators more freedom when defining axes.

Added legends to graphs. These can be defined through the 'legend' field in a module.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -17,13 +17,11 @@ define(['modules/moduleList'], function (dynModules) {
         function scaleAxis(module, object, axisname) {
             /*jshint maxcomplexity:7 */
             if(octopeerHelper.getSafeModuleValue(module,axisname+"AxisScale")() === "fit"){
-                var Scale = d3.scale.linear()
-                    .domain(octopeerHelper.getSafeModuleValue(module,axisname+"AxisFitFunction")())
-                    .range([350-50-10,0])
-                    .nice();
-                var Axis = d3.svg.axis().scale(Scale);
+                var Axis = octopeerHelper.getSafeModuleValue(module,axisname+"AxisFitFunction")();
```
Comment (borek2) [inline src/main.js]: only use capital letters as class names or constants, not variables

**Comment by borek2** [comment]
Small comment, apart from that nice job :)!

File: src/main.js (hunk 1 above)
Comment (agudek) [inline src/main.js]: Will fix with next additions.

Comment (agudek) [inline src/main.js]: @borek2 fixed

**Pull request merged**